src/sentry/region_to_control/producer.py

```diff
@@ -0,0 +1,33 @@
+import dataclasses
+from typing import Optional
+
+from django.conf import settings
+
+from sentry.models import UserIP
+from sentry.region_to_control.messages import Message, UserIpEvent
+from sentry.silo import SiloMode
+from sentry.utils import json, kafka_config
+from sentry.utils.pubsub import KafkaPublisher
+
+
+def produce_user_ip(event: UserIpEvent):
+    if SiloMode.get_current_mode() == SiloMode.REGION:
+        get_user_ip_publisher().publish(
+            settings.KAFKA_USER_IP_EVENTS,
+            json.dumps(dataclasses.asdict(Message(user_ip_event=event))),
+        )
+    else:
+        UserIP.objects.create_or_update(values=dataclasses.asdict(event))
+
+
+user_ip_publisher: Optional[KafkaPublisher] = None
+
+
+def get_user_ip_publisher() -> KafkaPublisher:
```

Comment:
This could use a comment explaining the motive for caching a global `KafkaPublisher` instance. As it is, it looks like we're just trying to avoid loading the options more than once. Is there more to it, like reusing the same connection or something like that?

If we're maintaining a global instance but initializing it lazily, I might prefer to make it as a `cached_property` of a singleton object. Otherwise, the global variable should be named `_user_ip_publisher` to indicate that people should call `get_user_ip_publisher()` instead of importing `user_ip_publisher`.